src/rgw/rgw_rest_pubsub.cc

```diff
@@ -1318,48 +1318,49 @@ void RGWPSCreateNotifOp::execute_v2(optional_yield y) {
     op_ret = -ERR_SERVICE_UNAVAILABLE;
     return;
   }
-
-  if (configurations.list.empty()) {
-    op_ret = remove_notification_v2(this, driver, s->bucket.get(),
-                                    /*delete all notif=true*/ "", y);
-    return;
-  }
-  rgw_pubsub_bucket_topics bucket_topics;
-  op_ret = get_bucket_notifications(this, s->bucket.get(), bucket_topics);
-  if (op_ret < 0) {
-    ldpp_dout(this, 1)
-        << "failed to load existing bucket notification on bucket: "
-        << s->bucket << ", ret = " << op_ret << dendl;
-    return;
-  }
-  for (const auto& c : configurations.list) {
-    const auto& notif_name = c.id;
-
-    const auto arn = rgw::ARN::parse(c.topic_arn);
-    if (!arn) { // already validated above
-      continue;
+  op_ret = retry_raced_bucket_write(this, s->bucket.get(), [&] {
```

Comment:
what capture everything by ref?
what do yo uactually need?

Reply:
> what capture everything by ref? what do yo uactually need?

it was just lazy coding, did not  bother checking the params needed
few places were just doing same of capturing all, so i did same.
will it matter ? or change and then we have to re run tuethology ?

Reply:
> will it matter

IMO, it mainly matters for the reader of the code, who needs to figure out what this lambda might change and/or need. but, to me, this is an important reason to be explicit as much as possible with capturing

Reply:
done